[RHACS] [Docs] ROX-34180: Add release notes for RHACS 4.10.2 (#110840)
openshift/openshift-docs: modules/bug-fixes-in-version-4102.adoc

// Module included in the following assemblies:
//
// * release_notes/410-release-notes.adoc

:_mod-docs-content-type: REFERENCE
[id="bug-fixes_4102_{context}"]
= Bug fixes in version 4.10.2

[role="_abstract"]
This release provides the following bug fixes:

//ROX-34014

> * Before this update, label search filters did not work correctly in the user interface. When users entered free-form text filters, the system did not properly handle label searches requiring special formatting for exact matches and regex patterns. With this release, Red{nbsp}Hat has corrected label search handling to properly format queries with equals signs and regex modifiers. As a result, label searches now work correctly for both exact matches and regex patterns.

Reviewer: ```suggestion
* Before this update, label search filters did not work correctly in the {product-title-short} portal. When users entered free-form text filters, the system did not properly handle label searches requiring special formatting for exact matches and regex patterns. With this release, Red{nbsp}Hat has corrected label search handling to properly format queries with equals signs and regex modifiers. As a result, label searches now work correctly for both exact matches and regex patterns.
```

//ROX-34000
* Before this update, the documentation for Central exposure route reencrypt TLS certificate and key configuration requirements was unclear. Users attempting to configure these fields encountered operator failures with the message "FATAL ERROR: The reencrypt route must specify either both, certificate and key, or neither." With this release, Red{nbsp}Hat has clarified the documentation to specify that users must configure the certificate and key fields together or omit both. As a result, users can now correctly configure reencrypt routes.

> //ROX-33977

Reviewer: I think this is TMI for most users and we don't want to expose the inner workings of the code as much.

Suggestion:

Before this update, the system calculated image component risk scores but did not store them, affecting deprecated dashboard displays. With this release, Red{nbsp}Hat has restored the risk score persistence functionality. As a result, image component risk scores are now properly stored and displayed.

Reviewer 2: I agree with your suggestion.

* Before this update, image component risk scores were not persisted after a code change removed the database write operation while still updating an in-memory map. This caused the system to calculate risk scores but not store them, affecting deprecated dashboard displays. With this release, Red{nbsp}Hat has restored the risk score persistence functionality. As a result, image component risk scores are now properly stored and displayed.

//ROX-33941

> * Before this update, the image store used an inefficient N+1 query pattern when fetching CVE data, issuing a separate database query for each component in an image. For images with 100 components, this resulted in 101 queries to load a single image, significantly impacting performance for batch operations, GraphQL queries, image search, and enrichment processes. With this release, Red{nbsp}Hat has optimized the CVE fetching to use bulk queries. As a result, image retrieval performance has improved significantly, especially for images with many components.

Reviewer: Again, TMI that exposes inner workings of the system.

Suggestion:

Before this update, the image store used an inefficient query pattern when fetching CVE data. This impacted performance for batch operations, GraphQL queries, image searches, and other actions for images with 100 or more components. With this release, Red{nbsp}Hat has optimized the CVE fetching, resulting in improved image retrieval performance, especially for images with many components.

Reviewer 2: I'm fine with what you wrote, but could we even reduce it further.  

Something like:
With this release, Red{nbsp}Hat has introduced some performance optimizations with CVE fetching, resulting in improved image retrieval performance, especially for images with many components.

//ROX-33935

> * Before this update, the image component ranker initialization process walked the entire image components table and deserialized full protobuf data just to extract ID and risk score fields. This caused excessive memory usage because zero-copy deserialization pinned entire row buffers in memory. In large deployments with about 15 million image components, this resulted in about 6.7 GB of retained heap memory and potential out-of-memory conditions. With this release, Red{nbsp}Hat has optimized the initialization to query only the required ID and risk score columns. As a result, memory usage during initialization has decreased by about 6.7 GB, reducing the risk of out-of-memory failures.

Reviewer: Again - way TMI for a user and exposes internal workings. Suggestion:

Before this update, the image component initialization process caused excessive memory usage in large deployments. With this release, Red{nbsp}Hat has optimized the initialization, decreasing memory usage  and reducing the risk of out-of-memory failures.

Reviewer 2: This one looks great.  The AI is really wordy.

//ROX-33934

> * Before this update, the system performed wasteful pruning operations on image component risk records that the system no longer wrote to the database. These costly searches consumed resources without finding any records to prune. With this release, Red{nbsp}Hat has removed the obsolete risk records and the associated pruning code. As a result, the system no longer performs unnecessary pruning operations, improving overall performance.

Reviewer: I'm not even sure this should be documented as it's an internal improvement and unlikely to be noticed by users except for the end result. Also, you then get into the customer questions of "what are you pruning, why are you pruning, what records are you removing from the system and why?" You could say something like:

```suggestion
* With this release, Red{nbsp}Hat removed obsolete pruning operations, improving overall performance.
```

Reviewer 2: Yeah I'm not sure this one needs to be in the release notes at all.  At most I would use your suggestion, but I think I would prefer to just remove it.

//ROX-33451

> * Before this update, the RHACS Operator performed unnecessary reconciliations due to two issues. The status controller reconciler triggered on deployment spec changes in addition to status changes, and the helm-reconciler predicate did not correctly filter out updates from the status controller when working with unstructured objects. With this release, Red{nbsp}Hat has corrected the status controller to trigger only on deployment status changes and fixed the predicate to work correctly with unstructured objects. As a result, the operator performs fewer unnecessary reconciliations, improving performance and reducing resource usage.

Reviewer: The user really only needs to know that we made changes to the Operator to improve performance. They likely do not care about reconciliations.

Suggestion:

```suggestion
* With this release, Red{nbsp}Hat made changes the Operator reconciliation to improve performance and reduce resource usage.
```

This release also addresses the following security vulnerabilities:

> * Security vulnerability in OpenTelemetry components (link:https://access.redhat.com/security/cve/CVE-2026-24051[CVE-2026-24051])

Reviewer: It is helpful to ask the AI tool to add the Jira ticket number as a commented line above each CVE fixed in case we need to look them up.

* Security vulnerability in Docker components (link:https://access.redhat.com/security/cve/CVE-2025-15558[CVE-2025-15558])
* Kubelet, CRI-O, kube-apiserver: Denial of service via SPDY streaming code (link:https://access.redhat.com/security/cve/CVE-2026-35469[CVE-2026-35469])

> * github.com/jackc/pgx: Memory-safety vulnerability (link:https://access.redhat.com/security/cve/CVE-2026-33815[CVE-2026-33815])

Reviewer: Combine and have the 2 CVE links at end separated by a comma?

* github.com/jackc/pgx: Memory-safety vulnerability (link:https://access.redhat.com/security/cve/CVE-2026-33816[CVE-2026-33816])

> * Go JOSE: Denial of service via crafted JSON Web Encryption (JWE) object (link:https://access.redhat.com/security/cve/CVE-2026-34986[CVE-2026-34986])

Reviewer: (Never mind)

Reviewer: Oops, my error, I see it now!

* gRPC-Go: Authorization bypass due to improper HTTP/2 path validation (link:https://access.redhat.com/security/cve/CVE-2026-33186[CVE-2026-33186])
* Immutable.js: Improperly controlled modification of object prototype attributes (prototype pollution) in immutable (link:https://access.redhat.com/security/cve/CVE-2026-29063[CVE-2026-29063])

> * Prototype pollution in _.unset and _.omit functions (link:https://access.redhat.com/security/cve/CVE-2025-13465[CVE-2025-13465])

Reviewer: This needs some asciidoc formatting to escape the variables.

Reviewer: This was actually in the 4.8.10 RNs and I fixed it this way:

```suggestion
* * lodash: Prototype pollution in `+++_.unset+++` and `+++_.omit+++` functions (link:https://access.redhat.com/security/cve/cve-2025-13465[CVE-2025-13465])
```
Might want to edit the instructions to the AI tool to look at RNs for the latest patches for 4.8, 4.9 to use previous content for CVE fixes listed if they already exist.
